PWGCF/TableProducer/dptdptfilter.cxx

```diff
@@ -0,0 +1,935 @@
+    }
+    fhTruePA[sp]->Fill(particle.p());
+    fhTruePtA[sp]->Fill(particle.pt());
+    if (charge > 0) {
+      fhTruePtPosA[sp]->Fill(particle.pt());
+    } else {
+      fhTruePtNegA[sp]->Fill(particle.pt());
+    }
+  }
+
+  template <typename TrackObject>
+  inline MatchRecoGenSpecies IdentifyTrack(TrackObject const& track)
+  {
+    using namespace o2::analysis::dptdptfilter;
+
+    float nsigmas[kDptDptNoOfSpecies];
+    if (track.p() < 0.8) {
+      nsigmas[kDptDptCharged] = 999.0f;
+      nsigmas[kDptDptElectron] = track.tpcNSigmaEl();
+      nsigmas[kDptDptMuon] = track.tpcNSigmaMu();
+      nsigmas[kDptDptPion] = track.tpcNSigmaPi();
+      nsigmas[kDptDptKaon] = track.tpcNSigmaKa();
+      nsigmas[kDptDptProton] = track.tpcNSigmaPr();
+    } else {
+      /* introduce require TOF flag */
+      if (track.hasTOF()) {
+        nsigmas[kDptDptCharged] = 999.0f;
+        nsigmas[kDptDptElectron] = sqrtf(track.tpcNSigmaEl() * track.tpcNSigmaEl() + track.tofNSigmaEl() * track.tofNSigmaEl());
+        nsigmas[kDptDptMuon] = sqrtf(track.tpcNSigmaMu() * track.tpcNSigmaMu() + track.tofNSigmaMu() * track.tofNSigmaMu());
+        nsigmas[kDptDptPion] = sqrtf(track.tpcNSigmaPi() * track.tpcNSigmaPi() + track.tofNSigmaPi() * track.tofNSigmaPi());
+        nsigmas[kDptDptKaon] = sqrtf(track.tpcNSigmaKa() * track.tpcNSigmaKa() + track.tofNSigmaKa() * track.tofNSigmaKa());
+        nsigmas[kDptDptProton] = sqrtf(track.tpcNSigmaPr() * track.tpcNSigmaPr() + track.tofNSigmaPr() * track.tofNSigmaPr());
+      } else {
+        nsigmas[kDptDptCharged] = 999.0f;
+        nsigmas[kDptDptElectron] = track.tpcNSigmaEl();
+        nsigmas[kDptDptMuon] = track.tpcNSigmaMu();
+        nsigmas[kDptDptPion] = track.tpcNSigmaPi();
+        nsigmas[kDptDptKaon] = track.tpcNSigmaKa();
+        nsigmas[kDptDptProton] = track.tpcNSigmaPr();
+      }
+    }
+    float min_nsigma = 999.0f;
+    MatchRecoGenSpecies sp_min_nsigma = kWrongSpecies;
+    for (int sp = 0; sp < kDptDptNoOfSpecies; ++sp) {
+      if (nsigmas[sp] < min_nsigma) {
+        min_nsigma = nsigmas[sp];
+        sp_min_nsigma = MatchRecoGenSpecies(sp);
+      }
+    }
+    bool doublematch = false;
+    if (min_nsigma < 3.0) {
+      for (int sp = 0; (sp < kDptDptNoOfSpecies) and not doublematch; ++sp) {
+        if (sp != sp_min_nsigma) {
+          if (nsigmas[sp] < 3.0) {
+            doublematch = true;
+          }
+        }
+      }
+      if (doublematch) {
+        return kWrongSpecies;
+      } else {
+        return sp_min_nsigma;
+      }
+    } else {
+      return kWrongSpecies;
+    }
+  }
+
+  template <typename ParticleObject>
+  inline MatchRecoGenSpecies IdentifyParticle(ParticleObject const& particle)
+  {
+    using namespace dptdptfilter;
+
+    constexpr int pdgcodeEl = 11L;
+    constexpr int pdgcodeMu = 13L;
+    constexpr int pdgcodePi = 211L;
+    constexpr int pdgcodeKa = 321L;
+    constexpr int pdgcodePr = 2212L;
```

Comment:
Why L if it is an int then?

Reply:
You are right! It should not be!
I will change them